### PR TITLE
Homework6: Igor Bohdan

### PR DESCRIPTION
**Pull request by IgorBohdan3407**

### CODE REVIEW

File: task_1_Homework6.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+
+class AddressBookEntry(object):
+    version = 1.0
+
+    def __init__(self, name, phone):
+        self.name = name
+        self.phone = phone
+
+    def __repr__(self):
```
Comment (belovmd) [inline task_1_Homework6.py]: why not str?

File: task_1_Homework6.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        return ' [Address Book Entry: %s, %s ] ' % (self.name, self.phone)
+
+    def updatePhone(self, phone):
```
Comment (belovmd) [inline task_1_Homework6.py]: not pythonic name

File: task_1_Homework6.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+        self.ssn = social
+
+
+class CustomerAddressBookEntry(AddressBookEntry):
+    def __init__(self, name, phone, order, date):  # Переопределить конструктор
+        self.order = order
+        self.date = date
+        AddressBookEntry.__init__(self, name, phone)
+
+    def updateOrder(self, order):
```
Comment (belovmd) [inline task_1_Homework6.py]: not pythonic

File: task_1_Homework6.py [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+print(robert.name, robert.phone)
+
+mitchel = CustomerAddressBookEntry('Makl Mitchel',
+                                   +375447496999,
+                                   'Order № 25',
+                                   '25.01.2020')
+
+print(mitchel)
+# "Магический" метод __repr__ в данном случае иллюстрирует
```
Comment (belovmd) [inline task_1_Homework6.py]: it's not polymorphism example.

File: task_1_Homework6.py [hunk 5]
```diff
@@ -0,0 +1,66 @@
+﻿""" Создайте  модель из жизни.
```
Comment (belovmd) [inline task_1_Homework6.py:1]: insufficient interaction in models

File: task_2_Homework6.py [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+def retry(max_tries):
+    def decorator(func):
+        @functools.wraps(func)
+        def wrapper(*args, **kwargs):
+            for n in range(1, max_tries + 1):
+                try:
+                    return func(*args, **kwargs)
+                except Exception:
+                    if n == max_tries:
+                        raise
```
Comment (belovmd) [inline task_2_Homework6.py]: it should raise TooManyErrors exception according to task description and not depend on decorated function implementation

**Review by belovmd — CHANGES_REQUESTED** [review]
Please take a look at these issues

**Review by belovmd — COMMENTED** [review]
please address previous review issues

File: task_1_Homework6.py (hunk 5 above)
Comment (IgorBohdan3407) [reply task_1_Homework6.py:1]: Please tell me what needs to be changed or added?

File: task_1_Homework6.py (hunk 4 above)
Comment (IgorBohdan3407) [reply task_1_Homework6.py]: Oops! I think I got excited)

File: task_2_Homework6.py [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+def retry(max_tries):
+    def decorator(func):
+        @functools.wraps(func)
+        def wrapper(*args, **kwargs):
+            for n in range(1, max_tries + 1):
+                try:
+                    return func(*args, **kwargs)
+                except Exception:
+                    if n == max_tries:
+                        raise IOError("TooManyErrors")
```
Comment (belovmd) [inline task_2_Homework6.py:23]: It should be exception class, not text for IOError

**Review by belovmd — COMMENTED** [review]
6,5
pretty trivial model
decorator solution not aligned with task description

**Pull request closed without merging**